src/EntityComponentManager.cc

```diff
@@ -493,7 +493,16 @@ Entity EntityComponentManager::CloneImpl(Entity _entity, Entity _parent,
     auto originalComp = this->ComponentImplementation(_entity, type);
     auto clonedComp = originalComp->Clone();
 
-    this->CreateComponentImplementation(clonedEntity, type, clonedComp.get());
+    auto updateData =
+      this->CreateComponentImplementation(clonedEntity, type, clonedComp.get());
+    if (updateData)
+    {
+      // When a cloned entity is removed, it erases all components/data so a new
+      // cloned entity should not have components to be updated
+      ignerr << "The component's data needs to be updated but this is not "
+             << "expected. Please submit an issue including your use case "
```

Comment:
What do you think about adding here the url the github repo ?

Reply:
Or alternatively, this could say something like "... not expected. This is an internal error."

I don't think we usually ask that users submit users, it's kind of implied. What do you think?

Reply:
bb7f37e0289cdde223571b2444a7be71594a55f0 I modified the error message to follow this format: https://github.com/gazebosim/gz-sim/blob/afd087055f87d3e4c199d8e786fbccd68d3c80c1/src/EntityComponentManager.cc#L1165-L1167